src/server/database/leveldb-adaptor.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const levelup = require('levelup')
+const PapanUtils = require('../utils.js')
+
+class LevelCollectionAdaptor {
+  constructor (db) {
+    this.db = db
+  }
+
+  put (doc) {
+    let db = this.db
+
+    return new Promise((resolve, reject) => {
+      db.put(doc._id, JSON.stringify(doc), (err) => {
```

Comment:
Paranoid developer note: JSON.stringify will fail if there are circular references in "doc".